### PR TITLE
Move prospectus logic for selecting runs here

### PR DESCRIPTION
**Pull request by bbaker6225**

### CODE REVIEW

File: course_discovery/apps/api/v1/tests/test_views/test_courses.py [hunk 1]
```diff
@@ -247,7 +247,7 @@ def test_list_key_filter(self):
         keys = ','.join([course.key for course in courses])
         url = '{root}?keys={keys}'.format(root=reverse('api:v1:course-list'), keys=keys)
 
-        with self.assertNumQueries(50):
+        with self.assertNumQueries(54):
```
Comment (bbaker6225) [inline course_discovery/apps/api/v1/tests/test_views/test_courses.py]: Queries haven't actually increased for the usual list/retrieve use case, just the search endpoint case since we don't have prefetching on any of our search models. Seems out of scope for this work

File: README.rst [hunk 2]
```diff
@@ -32,6 +32,12 @@ Is the build failing because translations are out of date?
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Run ``make check_translations_up_to_date`` and check in the generated *.mo & *.po files to your PR.
 
+Using elasticsearch locally
```
Comment (bbaker6225) [inline README.rst:35]: Unrelated, just wanted to add this to hopefully make this a little easier to find

File: course_discovery/apps/course_metadata/models.py [hunk 3]
```diff
@@ -531,6 +531,49 @@ def marketing_url(self):
...
+        """
+        Returns the course_runs relevant to the prospectus marketing site
+
+        Checks to make sure that every course run is marketable, and not
+        archived. Prospectus prefers that all the marketing runs are
+        enrollable, but if there are no marketable runs that are also
+        enrollable, we return the run closest to now from the other marketable
+        runs
+
+        Filtered and sorted in python to avoid additional calls to the database
```
Comment (jmyatt) [inline course_discovery/apps/course_metadata/models.py:556]: good call

**Review by jmyatt — APPROVED** [review]
looks good!
